src/tokens/saga.ts

```diff
@@ -206,9 +206,10 @@ export function* fetchTokenBalancesSaga() {
 }
 
 export function* tokenAmountInSmallestUnit(amount: BigNumber, tokenId?: string) {
-  const networkIds = Object.values(networkConfig.networkToNetworkId)
-  const tokens: TokenBalance[] = yield* select((state) => tokensListSelector(state, networkIds))
-  const tokenInfo = tokens.find((token) => token.tokenId === tokenId)
+  if (!tokenId) {
```

Comment:
I dislike when a function allows null/undefined in the function signature, but then throws a runtime error if it's actually called with null/undefined. It requires you to open up and read through every function to know if there is real null safety. (harder to understand, less safe from bugs)

it looks like all the consumers of this function have an entire `tokenInfo` object at their disposal.. maybe the "decimals" field from that could be taken as a parameter instead? And maybe throw one level higher if tokenInfo is null in `buildAndSendTx`? (the only place this is used where `tokenInfo` can currently be null)

Reply:
Yeah, you're totally right -- using `decimals` as an arg feels way cleaner to me.